### PR TITLE
CMCL-1424: InputAxisController can suppress input while camera is blending

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Raised by a user on the forum: https://forum.unity.com/threads/disabling-the-cinemachines-inputs-from-an-active-state-during-state-transitions.1412670/

The idea is to have an option to suppress user input (Orbital/PanTilt) when a camera is in a blend. 

Possible places to put the option:
- ~In the OrbitalFollow and PanTilt behaviours, as a checkbox~
- ~In the BlendHints, as a flag~
- In the InputAxisController (this was done)

image

### Testing status

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [x] Commented all public classes, properties, and methods
- [x] Updated user documentation

### Technical risk

low

### CODE REVIEW

**Review by AntoineCharton — APPROVED** [review]
That's a really great idea!

File: com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs [hunk 1]
```diff
@@ -808,5 +807,25 @@ protected void UpdateTargetCache()
...
         /// <returns>The Cinemachine component for that stage, or null if not present.</returns>
         public virtual CinemachineComponentBase GetCinemachineComponent(CinemachineCore.Stage stage) => null;
+
+        /// <summary>Returns true if this camera is currently live for some CinemachineBrain.</summary>
+        public bool IsLive => CinemachineCore.Instance.IsLive(this);
+
+        /// <summary>Check to see whether this camera is currently participating in a blend 
+        /// within its parent manager or in a CinemacineBrain</summary>
+        /// <returns>True if the camera is participating in a blend</returns>
+        public bool IsParticipatinInBlend()
```
Comment (gaborkb) [inline com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs:817]: typo participatin**G**

File: com.unity.cinemachine/Runtime/Core/InputAxisControllerBase.cs [hunk 2]
```diff
@@ -243,6 +250,10 @@ void Update()
             if (!Application.isPlaying)
                 return;
 
+            if (SuppressInputWhileBlending && TryGetComponent<CinemachineVirtualCameraBase>(out var vcam))
+                if (vcam.IsParticipatinInBlend())
+                    return;
```
Comment (gaborkb) [inline com.unity.cinemachine/Runtime/Core/InputAxisControllerBase.cs:255]: Could be just one if with 2 &&s. Might not matter.

```
if (SuppressInputWhileBlending && TryGetComponent<CinemachineVirtualCameraBase>(out var vcam) 
       && vcam.IsParticipatinInBlend()) 
       return;
```

**Review by gaborkb — APPROVED** [review]
minor / suggestions

File: com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs [hunk 3]
```diff
@@ -110,6 +110,18 @@ public enum StandbyUpdateMode
 
         // Cache for GameObject name, to avoid GC allocs
         string m_CachedName;
+        bool m_WasStarted;
+        bool m_SlaveStatusUpdated = false;
```
[the remark this answers is not in the document]
Comment (glabute) [reply com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs:114]: yes, good point, but that is for another PR

**Pull request merged**